include/babeltrace/compat/socket-internal.h

```diff
@@ -0,0 +1,395 @@
+#ifndef _BABELTRACE_COMPAT_SOCKET_H
+#define _BABELTRACE_COMPAT_SOCKET_H
+
+/*
+ * babeltrace/compat/socket.h
+ *
+ * Copyright (C) 2015-2017  Michael Jeanson <mjeanson@efficios.com>
+ *               2015  Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifdef __MINGW32__
+
+#include <winsock2.h>
+
+#define BT_INVALID_SOCKET INVALID_SOCKET
```

Comment:
The intention was not to namespace this compat code under bt_socket_ since we don't do it for any other compat wrappers. I guess I just suck at naming things, here is a proposal for a more structured naming scheme.

# Use socketlib for library initialisation, which is specific to windows
bt_socketlib_init()
bt_socketlib_fini()

# Use bt_socket_ for custom compat functions
bt_socket_interrupted()
bt_socket_errormsg()

# Use bt_ORIGFUNC_SOMETHING for wrapped functions
bt_close_socket()
bt_send_nosigpipe()

I would keep BT_SOCKET_ERROR and BT_INVALID_SOCKET because the first denotes a socket error return code and the second an invalide socket object id and not an invalide return code.

Thoughts?